### PR TITLE
implement CollectionUtils.sortByString

### LINKED ISSUE #62: Port CollectionUtils.sortByString
Port over the `CollectionUtils.sortByString` method from one of the client projects.

### PR DESCRIPTION
**Pull request by mrjones2014**
- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)
Fixes #62

### CODE REVIEW

File: src/utilities/collection-utils.ts [hunk 1]
```diff
@@ -182,6 +182,45 @@ const _replaceElementAt = <T>(
...
+    array.sort((a: T, b: T) => {
+        let aString = selector(a);
+        let bString = selector(b);
+
+        if (!caseSensitive) {
+            aString = aString.toLowerCase();
+            bString = bString.toLowerCase();
+        }
+
+        if (aString === "" || aString === null) {
```
Comment (brandongregoryscott) [inline src/utilities/collection-utils.ts:205]: Are we intentionally strict `null` checking, or should we also return here if `aString` is `undefined`?

File: src/utilities/collection-utils.test.ts [hunk 2]
```diff
@@ -572,4 +572,124 @@ describe("CollectionUtils", () => {
...
+            ];
+
+            // Act
+            const result = CollectionUtils.sortByString(array, selector);
+
+            // Assert
+            expect(result).toEqual(expected);
+        });
+
+        it("Puts empty string values at the end", () => {
```
Comment (brandongregoryscott) [inline src/utilities/collection-utils.test.ts:632]: Could this be reworded to better fit into Given/When/Then style test naming? `given there entries with empty string properties, it returns those values sorted at the end"

File: src/utilities/collection-utils.test.ts [hunk 3]
```diff
@@ -572,4 +572,124 @@ describe("CollectionUtils", () => {
...
+            ];
+
+            // Act
+            const result = CollectionUtils.sortByString(array, selector);
+
+            // Assert
+            expect(result).toEqual(expected);
+        });
+
+        it("Ignores case sensitivity when sorting", () => {
```
Comment (brandongregoryscott) [inline src/utilities/collection-utils.test.ts:608]: Same comment here - try to reword this to better fit G/W/T

File: src/utilities/collection-utils.ts [hunk 4]
```diff
@@ -182,6 +182,45 @@ const _replaceElementAt = <T>(
...
+    selector: (element: T) => string,
+    caseSensitive: boolean = false
+) =>
+    array.sort((a: T, b: T) => {
+        let aString = selector(a);
+        let bString = selector(b);
+
+        if (!caseSensitive) {
+            aString = aString.toLowerCase();
+            bString = bString.toLowerCase();
```
Comment (brandongregoryscott) [inline src/utilities/collection-utils.ts:202]: What happens if `aString` or `bString` is `null` or `undefined` here?

**Review by brandongregoryscott — CHANGES_REQUESTED** [review]
Left some questions/suggestions for renaming. Let me know when it's ready to re-review @mrjones2014 👍

File: src/utilities/collection-utils.test.ts [hunk 5]
```diff
@@ -572,4 +572,124 @@ describe("CollectionUtils", () => {
...
+
+    // -----------------------------------------------------------------------------------------
+    // #region sortByString
+    // -----------------------------------------------------------------------------------------
+
+    describe("#sortByString", () => {
+        type TestType = { letter: string };
+        const selector = (t: TestType) => t.letter;
+
+        it("Sorts an array by specified selector", () => {
```
Comment (wintondeshong) [inline src/utilities/collection-utils.test.ts:584]: Nitpick, please start these tests statements with lowercase for consistency and being it is a part of a larger constructed sentence

File: src/utilities/collection-utils.test.ts (hunk 3 above)
Comment (wintondeshong) [reply src/utilities/collection-utils.test.ts:608]: 👍

**Review by wintondeshong — CHANGES_REQUESTED** [review]
@mrjones2014 while no functional issues, please address @brandongregoryscott and my comments around wording and casing of the test methods when you have a moment. Thanks for the contribution!

File: src/utilities/collection-utils.ts (hunk 1 above)
Comment (mrjones2014) [reply src/utilities/collection-utils.ts:205]: oops 😬

File: src/utilities/collection-utils.ts (hunk 4 above)
Comment (mrjones2014) [reply src/utilities/collection-utils.ts:202]: addressing by using null-safe operator `?`

**Comment by mrjones2014** [comment]
@wintondeshong @brandongregoryscott comments should be addressed now

**Review by wintondeshong — APPROVED** [review]
Thanks for making those changes @mrjones2014

**Review by brandongregoryscott — APPROVED** [review]
